front/lib/api/assistant/conversation.ts

```diff
@@ -1998,22 +1988,24 @@ async function isMessagesLimitReached({
   plan: PlanType;
   subscription: SubscriptionType;
 }): Promise<boolean> {
+  const { maxMessages, maxMessagesTimeframe } = plan.limits.assistant;
+
   if (plan.limits.assistant.maxMessages === -1) {
     return false;
   }
 
   if (isTrial(subscription)) {
-    // For trials, the limit applies to the last 24 hours.
     const remaining = await rateLimiter({
-      key: `workspace:${owner.id}:agent_message_count:last_24_hours`,
-      maxPerTimeframe: 25,
-      timeframeSeconds: 60 * 60 * 24, // 1 day.
+      key: `workspace:${owner.id}:agent_message_count:${maxMessagesTimeframe}`,
+      maxPerTimeframe: maxMessages,
+      timeframeSeconds: getTimeframeSecondsFromLiteral(maxMessagesTimeframe),
       logger,
     });
 
     return remaining <= 0;
   }
 
+  // TODO(2024-03-18 flav) Should we keep this?
```

Comment:
@spolu what should we do about this code? Can we use Redis instead of the DB to have one path forward. The only downside that I see is that if Redis is restarted/crashed we will give another 50 messages to those legacy users, IMHO, it's not a big issue.

Reply:
We also use data in Redis forever which is not awesome?

Reply:
This will grow indefinitely. But at the same time this is only for free users which won't exist in the future... right?

Reply:
If my understanding is correct I'm fine relying on Redis. Maybe simply put a long expiry like 1month instead of lifetime?

Reply:
> This will grow indefinitely. But at the same time this is only for free users which won't exist in the future... right?

Correct, it's only for the workspaces with the legacy plan.

Reply:
Ok so I'll keep lifetime in the type but will map lifetime to a month in Redis.

Reply:
Fixed.